pkg/reconciler/v1alpha1/testing/functional.go

```diff
@@ -250,6 +250,16 @@ func WithFailedConfig(name, reason, message string) ServiceOption {
 	}
 }
 
+// WithRouteNotReady sets the `RoutesReady` condition on the service to `Unknown`.
+func WithRouteNotReady(s *v1alpha1.Service) {
```

Comment:
This function could use a better name and comment as it is distinctly different, but could be easily confused with the initialization condition where Route.Ready=Unknown and Service.RouteReady=Unknown versus the condition you are setting here with Route.Ready=True and Service.RouteReady=Unknown.

Reply:
Renamed.